internal/datastore/spanner/watch.go

```diff
@@ -102,10 +106,22 @@ func (sd spannerDatastore) loadChanges(
 			&tpl.Subject.Namespace,
 			&tpl.Subject.ObjectId,
 			&tpl.Subject.Relation,
+			&caveatName,
+			&caveatCtx,
 		)
 		if err != nil {
 			return err
 		}
+		if caveatName.Valid && caveatCtx.Valid {
```

Comment:
hrm, why checking `caveatCtx.Valid` though? It's entirely possible to have a caveated relationship with a null context. May be worth checking if our tests exercise getting updates for both relationships with caveat but null context and caveat with non-null context